code/app/auth/views.py

```diff
@@ -67,6 +67,46 @@ def login():
             flash('Invalid email or password.', 'error')
     return render_template('security/login_user.html', form=form)
 
+#Admin route
+@auth.route('/admin', methods=['GET', 'POST'])
+@login_required
+@permission_required('admin')
+def admin():
+    nrole = NewRoleForm()
+    if nrole.create.data and nrole.validate():
+        if not Role.query.filter_by(name=nrole.name.data).first():
+            print(nrole.name.data)
+            new_role = Role(name=nrole.name.data)
+            db.session.add(new_role)
+            db.session.commit()
+            flash("Role created.",'success')
+        else:
+            flash("Role already exists.",'error')
+
+    #Deal with requests to assign roles to users
+    assign = AssignRoleForm()
+    if assign.assign.data and assign.validate():
+        for u in assign.user.iter_choices():
+            if u[2] == True:
+                user = User.query.get(u[0])
+                for r in assign.new_roles.iter_choices():
+                    if r[2] == True:
+                        user.assign_role(r[1])
+
+    #Deal with requests to remove roles from users
+    remove = RemoveRoleForm()
+    if remove.remove.data and remove.validate():
+        for u in remove.user.iter_choices():
+            if u[2] == True:
+                user = User.query.get(u[0])
+                for r in remove.rem_roles.iter_choices():
+                    if r[2] == True:
+                        print(r[1])
+                        user.remove_role(r[1])
+
+    return render_template('security/admin.html', nrole=NewRoleForm(),
```

Comment:
You are referencing `security/admin.html` but it does not exist. I get Error: `jinja2.exceptions.TemplateNotFound: security/admin.html`
Did you miss adding the files in the `security` directory?

Reply:
That's exactly what happened!  I've added it and pushed to origin, it's there now.

Reply:
FYI another branch just merged into master and I updated to allow the merge, so it's worth a double check on my content.

Reply:
Was working before the merge. Not sure why is not working now.

Reply:
I'll take a look

Reply:
Ok, think I fixed it.  I don't even see a merge so I don't know where that came from at all.  A git pull from master added a lot of stuff I hadn't seen before.